test/com/facebook/buck/jvm/java/testdata/javac_plugin/BUCK.fixture

```diff
@@ -0,0 +1,45 @@
+java_plugin(
```

Comment:
So we're calling the rule `java_plugin`, though we refer to it in code as Java**c**Plugin. The fields on this rule also don't match Bazel's [`java_plugin`](https://docs.bazel.build/versions/master/be/java.html#java_plugin), which looks to be for annotation processors only. @styurin, do you have a preference on the rule name (or API more generally) here?

Reply:
"Bazel's `java_plugin` defines plugins for the Java compiler run by Bazel. At the moment, the only supported kind of plugins are annotation processors."

Buck is already different as we have `java_annotation_processor` rule, also with fields that are very different from bazel's `java_plugin` rule.

Bazel's java_plugin rule accepts srcs and a bunch of other things that just couples it to a `java_library` rule instead of only defining it as a javac plugin, as buck does with `java_annotation_processor` and now `java_plugin`.

This way the API is decoupled from existing code while serving the same purpose.

Also, the idea in separating `java_plugin` from `java_annotation_processor` is that they are different, for example, it's common to have many processors in a single annotation processor, but not on plugins.
The [JavacTask](https://docs.oracle.com/javase/7/docs/api/javax/tools/JavaCompiler.CompilationTask.html?is-external=true#setProcessors(java.lang.Iterable)) provides an API to accept processors, but not plugins, etc...
So decoupling them makes sense as they are not the same.

Reply:
My question was not about whether or not we should separate `java_plugin` and `java_annotation_processor` (we should), but rather whether or not `java_plugin` is a good name, vs. for example `javac_plugin`. I don't have a strong opinion here. `java_plugin` is probably fine, but has the downside of not being what a Bazel user might expect it to be.

Reply:
I see, let me know what you guys think it's best. IMO both names are good enough.

Reply:
`java_plugin` is ok by me.